pkg/kubelet/volumes.go

```diff
@@ -77,12 +77,18 @@ func (kl *Kubelet) mountExternalVolumes(pod *api.BoundPod) (volumeMap, error) {
 	for i := range pod.Spec.Volumes {
 		volSpec := &pod.Spec.Volumes[i]
 
+		podRef, err := api.GetReference(pod)
```

Comment:
Sorry for not being clear-- I meant "should" as in "it is desirable but would require some work before it is possible". :)

api.BoundPod has a spot for a SelfLink. We should fill that in when we read a pod from config file.

Or--and this would make @erictune your friend for life--we could make GetReference not fail when the SelfLink is blank.